cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/DashboardPage.java

```diff
@@ -0,0 +1,269 @@
+    private WebElement tableChecking;
+
+    @FindBy(id = ACCORDION_SAVINGS_TABLE_ID)
+    private WebElement tableSavings;
+
+    @FindBy(id = ACCORDION_CREDIT_CARDS_TABLE_ID)
+    private WebElement tableCreditCards;
+
+    @FindBy(id = ACCORDION_INVESTMENTS_TABLE_ID)
+    private WebElement tableInvestments;
+
+    @FindBy(id = TRANSACTION_ARROW_OVERLAY_ID)
+    private WebElement transactionArrowOverlay;
+
+    //Web Elements - Transaction Arrow Buttons
+    @FindBy(id = BTN_TRANSCATION_ARROWS_ID)
+    private List<WebElement> btnTransactionButtons;
+
+
+    //Web Elements - Charts
+    @FindBy(id = CURRENT_SPENDING_CHART_TITLE_ID)
+    private WebElement currentSpendingChartTitle;
+
+    @FindBy(id = CURRENT_SPENDING_CHART_ID)
+    private WebElement currentSpendingChart;
+
+    @FindBy(id = BUDGET_CIRCLE_CHART_ID)
+    private WebElement budgetCircleChart;
+
+    @FindBy(id = BUDGET_BREAKDOWN_ID)
+    private List<WebElement> budgetBreakdownItems;
+
+    //Child Components
+    DashboardNavbar navbar;
+
+//#END Attributes
+
+    /////////// CONSTRUCTORS //////////////////////////   
+    public DashboardPage(WebDriver driver) {
+        super(driver);
+        //initialize WebElements
+        PageFactory.initElements(driver, this);
+
+        //load components
+        navbar = new DashboardNavbar(driver);
+
+        //Map WebElements
+        nameElementMap.put(NAME_ACCORDION_CHECKING_BTN, btnAccordianChecking);
+        nameElementMap.put(NAME_ACCORDION_SAVINGS_BTN, btnAccordianSavings);
+        nameElementMap.put(NAME_ACCORDION_CREDIT_CARDS_BTN, btnAccordianCreditCards);
+        nameElementMap.put(NAME_ACCORDION_INVESTMENTS_BTN, btnAccordianInvestments);
+        nameElementMap.put(ACCORDION_CHECKING_TABLE_ID, tableChecking);
+        nameElementMap.put(ACCORDION_SAVINGS_TABLE_ID, tableSavings);   
+        nameElementMap.put(ACCORDION_CREDIT_CARDS_TABLE_ID, tableCreditCards);  
+        nameElementMap.put(ACCORDION_INVESTMENTS_TABLE_ID, tableInvestments); 
+        nameElementMap.put(NAME_TRANSACTION_ARROW_OVERLAY, transactionArrowOverlay);
+        nameElementMap.put(NAME_CURRENT_SPENDING_CHART_TITLE, currentSpendingChartTitle);
+        nameElementMap.put(NAME_CURRENT_SPENDING_CHART, currentSpendingChart);
+        nameElementMap.put(NAME_BUDGET_CIRCLE_CHART, budgetCircleChart);
+        nameElementMap.put(NAME_BUDGET_BREAKDOWN, budgetBreakdownItems.get(0));
+        nameElementMap.put(NAME_TRANSACTION_ARROW, btnTransactionButtons.get(0)); //just the first arrow button
+        //NOTE: btnTransactionButtons are a list, so that needs to be accounted for in "get" methods
+    }
+
+    ////////// UNIQUE METHODS ////////////////////
+
+    /**
+     * Retrieves the table name associated with a button name.
+     * @param btnName - name of the button to retrieve the table name for.
+     * @return - name of the table associated with the button.
+     */
+    public String getTableNameFromButtonName(String btnName) {
```

Comment:
Question: button names have typically been tied to Gherkin files, but the table name returned is the React ID of the table tag. Is this meant to get the table name tied to the Gherkin file or is this correct?

Reply:
haven't played around with this for a while, and the naming system i used to start with was terrible, fixing it now

Reply:
modified to have better naming convention and to do what I wanted it to do in the first place. Thanks for the catch!